finetune/utils.py

```diff
@@ -412,3 +427,169 @@ def indico_to_finetune_sequence(texts, labels=None, none_value=config.PAD_TOKEN)
         all_subseqs.append(doc_subseqs)
         all_labels.append(doc_labels)
     return all_subseqs, all_labels
+
+
+# The following functions relating to the recomputation of gradients was taken and modified from tensor2tensor.
+
+def fn_with_custom_grad(grad_fn, use_global_vars=False):
```

Comment:
i would just move this to a different file.  we should talk about eventually breaking up utils.py up into logical files.  utils.py is kind of an antipattern